### PR TITLE
Exclude profiler

### PR DESCRIPTION
**Pull request by gmarler**
Add option `--exclude-self` that will prevent profiling the profiler.

Updated test for the new option.

### CODE REVIEW

File: src/profiler.rs [hunk 1]
```diff
@@ -1216,6 +1222,10 @@ impl Profiler<'_> {
     }
 
     fn should_profile(&self, pid: i32) -> bool {
+        if self.exclude_self && (pid == std::process::id().try_into().unwrap()) {
```
Comment (javierhonduco) [inline src/profiler.rs:1225]: nit: no need for the parenthesis, and as we know that the type fits, we can use `as` until we have our own type

```suggestion
        if self.exclude_self && pid == std::process::id() as i32 {
```

File: src/profiler.rs [hunk 2]
```diff
@@ -398,6 +402,7 @@ impl Profiler<'_> {
             .lightswitch_config
             .verbose_logging
             .write(profiler_config.bpf_logging);
+        let exclude_self = profiler_config.exclude_self;
```
Comment (javierhonduco) [inline src/profiler.rs:405]: nit: no need to extract to a variable if we only use it here

**Review by javierhonduco — APPROVED** [review]
LGTM! Left two small nits

File: src/profiler.rs (hunk 2 above)
Comment (gmarler) [reply src/profiler.rs:405]: We do use it in the Profiler initialization

File: src/profiler.rs (hunk 1 above)
Comment (gmarler) [reply src/profiler.rs:1225]: Fixed

**Pull request merged**